src/seedu/addressbook/logic/Logic.java

```diff
@@ -85,7 +85,9 @@ public CommandResult execute(String userCommandText) throws Exception {
     private CommandResult execute(Command command) throws Exception {
         command.setData(addressBook, lastShownList);
         CommandResult result = command.execute();
-        storage.save(addressBook);
+        if(command.isMutating()){
```

Comment:
Good that you've updated the functionality of the execute command. Since you've changed the implementation, take note to always update the header comment for the method as well, because the behaviour of executing changes due to the way data is saved.